sdk/security_keyvault/src/clients/keyvault_client.rs

```diff
@@ -54,63 +58,51 @@ impl KeyvaultClient {
     ) -> azure_core::Result<Self> {
         let vault_url = Url::parse(vault_url)?;
```

Comment:
Can we eventually adopt something like `azure_data_cosmos::CloudLocation` instead of dealing in raw URLs?